cirq-google/cirq_google/devices/serializable_device.py

```diff
@@ -101,15 +101,22 @@ def __init__(
         self.qubits = qubits
         self.gate_definitions = gate_definitions
         has_subcircuit_support: bool = cirq.FrozenCircuit in gate_definitions
+
         self._metadata = cirq.GridDeviceMetadata(
-            qubit_pairs=[
-                (pair[0], pair[1])
-                for gate_defs in gate_definitions.values()
-                for gate_def in gate_defs
-                if gate_def.number_of_qubits == 2
-                for pair in gate_def.target_set
-                if len(pair) == 2 and pair[0] < pair[1]
-            ],
+            qubit_pairs=cast(
+                List[Tuple[cirq.GridQubit, cirq.GridQubit]],
+                [
+                    (pair[0], pair[1])
```

Comment:
put the cast here so you're casting less things? Like it still can infer the `List[Tuple[...` part; you need to cast the qubits themselves. Although with the `isinstance` checks I'm sad that that's necessary, see below.